### PR TITLE
Using token in addition to secure_key in cron (36176)

### PR DESCRIPTION
**Pull request by bogdan202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Using token in addition to secure_key in cron
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes 36176
| How to test?  | Load the front crontroller with one token

### CODE REVIEW

File: controllers/front/syncAll.php [hunk 1]
```diff
@@ -56,6 +56,22 @@ public function __construct()
...
+        if (Tools::getValue('secure_key') == $this->module->secure_key) {
+            return true;
+        }
+
+        if (false == empty((new ShoppingfeedToken())->findByToken(Tools::getValue('token', '')))) {
+            return true;
+        }
+
+        $return = ['success' => false, 'error' => 'Authentication failed'];
+        $this->ajaxDie(json_encode($return));
```
Comment (clotairer) [inline controllers/front/syncAll.php:72]: ajaxDie not available on 1.6.0. has SF module a Backward compatibility on that ?

**Review by clotairer — CHANGES_REQUESTED** [review]

File: controllers/front/syncAll.php (hunk 1 above)
Comment (bogdan202) [reply controllers/front/syncAll.php:72]: There is no compatibility.
Pay attention, this method is used on the level of classlib
https://github.com/shoppingflux/module-prestashop/blob/develop/vendor/totpsclasslib/src/Extensions/ProcessMonitor/Controllers/Front/CronController.php#L80

So, I find it is better to provide compatibility on the lever of classlib.
At least on the branch, we use 'feature/34626_diagnostic'

Comment (clotairer) [reply controllers/front/syncAll.php:72]: You are right. There is an issue on this piece of classlib code.
'feature/34626_diagnostic' is a none stable version.

Could you fix your part of code. I'll talk to classlib team to fix to in the right branch.

Comment (bogdan202) [reply controllers/front/syncAll.php:72]: I have already fixed this problem one year ago
https://github.com/shoppingflux/module-prestashop/blob/develop/classes/ShoppingfeedCronController.php#L33
Sorry for confusing you

**Review by clotairer — APPROVED** [review]

**Pull request merged**